src/main/java/christmas/domain/Order.java

```diff
@@ -0,0 +1,100 @@
+package christmas.domain;
+
+import christmas.exception.ErrorMessage;
+import christmas.exception.InvalidOrderException;
+import christmas.global.Const;
+import christmas.global.Format;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Order {
+    private Date date;
+    private Map<Menu, Integer> order;
+
+    private Order(Date date, Map<Menu, Integer> order) {
+        validateQuantities(order);
+        validateMenuType(order);
+        this.date = date;
+        this.order = order;
+    }
+
+    public static Order of(Date date, Map<String, Integer> userOrder) {
+        Map<Menu, Integer> order = userOrder.entrySet().stream()
+                .collect(Collectors.toMap(
+                        userMenu -> Menu.findMatchMenu(userMenu.getKey()),
+                        Map.Entry::getValue
+                ));
+
+        return new Order(date, order);
+    }
+
+    private void validateQuantities(Map<Menu, Integer> order) {
+        int totalQuantities = calculateTotalQuantity(order);
+
+        if (isOverLimit(totalQuantities)) {
+            throw InvalidOrderException.of(ErrorMessage.OVER_LIMIT);
+        }
+    }
+
+    private int calculateTotalQuantity(Map<Menu, Integer> order) {
+        return order.values().stream().mapToInt(Integer::intValue).sum();
+    }
+
+    private boolean isOverLimit(int quantities) {
+        return quantities > Const.LIMIT_QUANTITY;
+    }
+
+    private void validateMenuType(Map<Menu, Integer> order) {
+        if (isOnlyDrinkType(order)) {
+            throw InvalidOrderException.of(ErrorMessage.NOT_EXIST_VALUE);
+        }
+    }
+
+    private boolean isOnlyDrinkType(Map<Menu, Integer> order) {
+        return order.isEmpty() && order.keySet().stream()
+                .allMatch(menu -> menu.getType() == Type.Drinks);
+    }
```

Comment:
(요구사항과 달라요!)

위 코드에서 쇼트서킷으로 인해서 order가 비어있지만 않으면 참으로 동작합니다.
![image](https://github.com/kkonii/java-christmas-6-kkonii/assets/73726591/72ff550f-5191-443c-b521-859f5f121801)
```suggestion
    private boolean isOnlyDrinkType(Map<Menu, Integer> order) {
        return order.isEmpty() || order.keySet().stream()
                .allMatch(menu -> menu.getType() == Type.Drinks)
    }
```
위 코드로 변경해주면 정상적으로 동작합니다 :)